client-device-connection-infinispan/src/main/java/org/eclipse/hono/deviceconnection/infinispan/client/AdapterInstanceStatusProvider.java

```diff
@@ -0,0 +1,44 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.deviceconnection.infinispan.client;
+
+import org.eclipse.hono.util.AdapterInstanceStatus;
+
+/**
+ * Provides the status of an adapter instance.
+ */
+public interface AdapterInstanceStatusProvider {
+
+    AdapterInstanceStatusProvider UNKNOWN_STATUS_PROVIDER = new UnknownStatusProvider();
+
+    /**
+     * Gets the status of the adapter identified by the given identifier.
+     *
+     * @param adapterInstanceId The identifier of the adapter.
+     * @return The status of the adapter instance.
+     * @throws NullPointerException if adapterInstanceId is {@code null}.
+     */
+    AdapterInstanceStatus getStatus(String adapterInstanceId);
```

Comment:
How about returning a `Future<AdapterInstanceStatus>` here instead in order to also support asynchronous implementations?

Reply:
I initially had used a `Future` return value here, but then didn't use it because the implementation in this PR doesn't need it and the code is simpler without it. An implementation that isn't doing time-consuming things here that require asynchronous decoupling would be preferred anyway, since this method is invoked for every command message.
And if we really need a Future at some point here, I think we can still change it then.